Actions/ReadSecrets/ReadSecretsHelper.psm1

```diff
@@ -170,9 +184,13 @@ function GetKeyVaultSecret {
     if (-not $script:keyvaultConnectionExists) {
             
         InstallKeyVaultModuleIfNeeded
-            
-        $credentialsJson = Get-KeyVaultCredentials
-        ConnectAzureKeyVaultIfNeeded -subscriptionId $credentialsJson.subscriptionId -tenantId $credentialsJson.tenantId -clientId $credentialsJson.clientId -clientSecret $credentialsJson.clientSecret
+
+        if ($script:gitHubSecrets.PSObject.Properties.Name -eq "AZURE_KEYVAULT_URI") {
```

Comment:
Same here - the AZURE_CREDENTIALS will stop working if people create a secret called AZURE_KEYVAULT_URI - or do I misread the code?

Reply:
That is correct - If they've created the other secrets as well (AZURE_CLIENT_ID, AZURE_CLIENT_SECRET, AZURE_KEYVAULT_SUBSCRIPTION_ID, AZURE_TENANT_ID) they shouldn't need AZURE_CREDENTIALS anymore. 

If we're worried about people having a AZURE_KEYVAULT_URI secret in their repo already (for another purpose) we can consider adding "ALGO_" as a prefix to the secrets?

Reply:
I am worried that if people will setup signing - then everything breaks until they have moved everything over.
If you have 100 repositories and create the signing secrets as org. secrets - then all repos might break depending on setup.

Reply:
It seems like we are adding 5 new secrets to replace 1 existing secret - and once you create them - the old one doesn't work any more. If you only support one keyvault - there should only be one way to setup the connection IMO.

Reply:
In that case of someone owning 100 Al-GO repositories I would assume the AZURE_CREDENTIALS is already set up as an organizational secret. In that case, they'd only need to remove that one secret and add the 5 new ones. Nothing should break in that scenario. 

It's not that the once you add the 5 secrets, the old mechanism doesn't work anymore. It shouldn't be needed anymore and should be removed. There would be no reason to keep both authentication mechanisms since they accomplish the same goal.

Reply:
If they are using the same keyvault for all repos - yes - but that isn't a requirement.
As I understand it - once you add the 5 secrets - AZURE_CREDENTIALS stops working.
Lets talk tomorrow at the sync meeting once more and land this.